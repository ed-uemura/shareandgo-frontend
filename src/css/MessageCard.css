.new-message {
	color: var(--SGGreen);
}

.new-message span {
	color: var(--SGGreen);
	font-size: var(--SGTextDetails);
}

.new-message .material-symbols-outlined {
	font-variation-settings: 'FILL' 1, 'wght' 400, 'GRAD' 0, 'opsz' 48;
}
