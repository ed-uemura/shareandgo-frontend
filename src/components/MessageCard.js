import React from 'react';
import { Link } from 'react-router-dom';
import '../css/MessageCard.css';

const MessageCard = ({
	driverPhoto,
	driverName,
	destination,
	date,
	time,
	link,
}) => {
	return (
		<Link to={link}>
			<div className='ride-card'>
				<div className='photo'>
					<figure className='driver-picture'>
						<img src={driverPhoto} alt={driverName} />
					</figure>
				</div>
				<div className='driver-info'>
					<p className='text-important'>{driverName}</p>

					<div className='destination d-flex align-items-center'>
						<span className='material-symbols-outlined card-icon me-2'>
							directions_car
						</span>
						<p className='details'>{destination}</p>
					</div>
				</div>
				<div className='ride-info'>
					<p className='details new-message'>
						<span className='material-symbols-outlined me-1'>circle</span>
						New
					</p>
					<p className='ride-details'>{date}</p>
					<p className='ride-details'>{time}</p>
				</div>
			</div>
		</Link>
	);
};

export default MessageCard;
