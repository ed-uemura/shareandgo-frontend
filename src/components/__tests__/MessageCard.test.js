import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import MessageCard from '../MessageCard';
import DriverImage from '../../images/profile-picture-mockup.jpg';
import { MemoryRouter } from 'react-router-dom';


afterEach(() =>{
    cleanup();
});

test('matches snapshot', () =>{ 
    const testInfo ={
    driverPhoto : DriverImage,
	driverName : 'Mathew Pitts',
	destination : 'Abbotsford, BC',
	date : '10/15/2022',
	time : '9:45am',
	link : '/message-content',
    }
    const tree = renderer.create(
        <MemoryRouter>
            <MessageCard {...testInfo}/>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});