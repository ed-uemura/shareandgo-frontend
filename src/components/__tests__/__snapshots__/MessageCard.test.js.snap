// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  href="/message-content"
  onClick={[Function]}
>
  <div
    className="ride-card"
  >
    <div
      className="photo"
    >
      <figure
        className="driver-picture"
      >
        <img
          alt="Mathew Pitts"
          src="profile-picture-mockup.jpg"
        />
      </figure>
    </div>
    <div
      className="driver-info"
    >
      <p
        className="text-important"
      >
        Mathew Pitts
      </p>
      <div
        className="destination d-flex align-items-center"
      >
        <span
          className="material-symbols-outlined card-icon me-2"
        >
          directions_car
        </span>
        <p
          className="details"
        >
          Abbotsford, BC
        </p>
      </div>
    </div>
    <div
      className="ride-info"
    >
      <p
        className="details new-message"
      >
        <span
          className="material-symbols-outlined me-1"
        >
          circle
        </span>
        New
      </p>
      <p
        className="ride-details"
      >
        10/15/2022
      </p>
      <p
        className="ride-details"
      >
        9:45am
      </p>
    </div>
  </div>
</a>
`;
